kythe/cxx/indexer/cxx/IndexerASTHooks.cc

```diff
@@ -1030,6 +1049,20 @@ bool IndexerASTVisitor::TraverseDecl(clang::Decl* Decl) {
   if (Decl == nullptr) {
     return true;
   }
+
+  // This is a dirty, dirty hack to allow other parts of the code to more
+  // cleanly handle deduced types.
```

Comment:
It seems like this is a departure from the way this was handled previously, or maybe you're solving a different problem. Can you provide an example of what this is intended to fix? Are you sure that it's safe to use overrideType here? In particular, if we make calls into Sema, will Clang ever inspect the replaced value?

Reply:
This should generally be safe, in that the only times these differ are around the deduced-ness of the types in question and temporarily swapping in one with the other shouldn't introduce any meaningful differences. This does include IncompleteArrayType vs ConstantArrayTypes, however, but for our purposes this should also be fine.